test/e2e/aadpodidentity_test.go

```diff
@@ -573,7 +590,7 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 		Expect(ok).To(Equal(true))
 
 		// setup mic and nmi with old releases
-		setupInfra("mcr.microsoft.com/k8s/aad-pod-identity", "1.4", "1.3")
+		setupInfraOld("mcr.microsoft.com/k8s/aad-pod-identity", "1.4", "1.3")
```

Comment:
I think instead of 2 functions we should just pass a boolean and use a single template to generate the final deployment.

Reply:
Did not want to make changes in setupinfra all through, instead kept the changes to internal function only.

Reply:
Ideally we should generate the required template based on version, <= 1.4. When that change is made in future PR, this single instance and setupold function will go away.